_includes/nav/current.html

```diff
@@ -6,6 +6,13 @@
   {% assign _current = true %}
 {% elsif page.collection and include.page.subnav.type == page.collection %}
   {% assign _current = true %}
+<!--
+  Use the page's `type` data key to connect the sidenav to a collection.
+  This sets "current" for the pages inside the _posts directory.
+  Collections are defined in config.yml, and "posts" is a default collection type in Jekyll.
+-->
+{% elsif page.collection and include.page.type == page.collection %}
+  {% assign _current = true %}
```

Comment:
## Needs discussion
This change allows us to set the "All news and updates" page as current in the side nav for all of our posts. To connect the pages, we needed to also add `type: posts` to the front matter of the What's new pages.

Note: `type` is an established data key in the front matter across the site, but `type: posts` is not yet used in the repo. 

### Why I didn't use the existing `subnav.type` check
Currently, the site relies on `subnav.type = "posts"` on the `updates.md` file to connect the posts to the "News and updates" page in the side navigation. 

However, I ran into some errors with this approach when I tried to add both a `type` and a link to "All news and upadtes" inside of `subnav` in `pages/whats-new/overview.md`. It seemed to create a conflict that caused the side navigation to break (But maybe I am just doing it wrong?).

### Proposed additional change
I recommend we remove the `subnav.type` check in lines 7-8 and replace it with the check added in lines 14-15. 

Currently, there is only one other page that declares a `subnav.type`: [security.md](https://github.com/uswds/uswds-site/blob/main/pages/about/security.md?plain=1#L8). Replacing `subnav.type` with `type` here should still connect it to the expected collection in [config.yml](https://github.com/uswds/uswds-site/blob/main/_config.yml#L77).

I don't see anything in the repo relating to `page.type` that would cause any regressions, but of course we would want to test that. 

Curious to hear thoughts on this approach and proposed additional change.

Reply:
This is somewhere I was also getting caught up while testing. I think it would be a good idea to normalize this across pages. It would also reduce complexity of this current page logic which would be a win!

**Praise:** Thank you for breaking this down in a clear and easy to follow way!

Reply:
Update: I opened issue #2974 to explore this conversation so that we can prevent scope creep.